### PR TITLE
emacsPackages.codeium: rework

### PR DESCRIPTION
**Pull request by AndersonTorres**
- set codeium explicitly instead of relying on pkgs
- rename patch to a more descriptive <num>-<description>.patch format
- use lib.getExe' instead of hardcoding the binary path
- updateScript

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/emacs/elisp-packages/manual-packages/codeium/default.nix [hunk 1]
```diff
@@ -19,17 +20,19 @@ melpaBuild {
...
   patches = [
     (substituteAll {
-      src = ./codeium.el.patch;
-      codeium = "${pkgs.codeium}/bin/codeium_language_server";
+      src = ./0000-set-codeium-command-executable.patch;
+      codeium = lib.getExe' codeium "codeium_language_server";
     })
   ];
 
+  passthru.updateScript = unstableGitUpdater { };
```
Comment (jian-lin) [inline pkgs/applications/editors/emacs/elisp-packages/manual-packages/codeium/default.nix:28]: This packages uses a stable release.  So `unstableGitUpdater` does not work.  You can use `gitUpdater` instead.

File: pkgs/applications/editors/emacs/elisp-packages/manual-packages/codeium/default.nix [hunk 2]
```diff
@@ -1,9 +1,10 @@
 {
+  lib,
+  codeium,
   fetchFromGitHub,
   melpaBuild,
-  pkgs,
-  lib,
```
Comment (jian-lin) [inline pkgs/applications/editors/emacs/elisp-packages/manual-packages/codeium/default.nix:5]: Ideally, we should not move parameters around.

Comment (AndersonTorres) [reply pkgs/applications/editors/emacs/elisp-packages/manual-packages/codeium/default.nix:5]: `lib` is tipically the first parameter around Nixpkgs.

Further, `pkgs` was deleted.

**Comment by jian-lin** [comment]
If there are no conflict with base branch (master here) to resolve, there is not need to sync with master again when you force-push.  In that way, it is easier to view the diff of a force-push by clicking the "Compare" button.

**Review by jian-lin — APPROVED** [review]

**Review by jian-lin — APPROVED** [review]
Since it uses a stable version, presumably adding an update script does not adds much extra maintenance burden to the maintainer.

**Pull request merged**